src/SegmentedString.chpl

```diff
@@ -1112,6 +1117,59 @@ module SegmentedString {
       return ranks;
     }
 
+    proc getFixesHelper(n: int, kind: Fixes, proper: bool) {
+      const lengths = getLengths() - 1;
+      var longEnough = makeDistArray(size, bool);
+      if proper {
+        longEnough = (lengths > n);
+      } else {
+        longEnough = (lengths >= n);
+      }
+      var nFound = + reduce longEnough;
+      var retOffsets = makeDistArray(nFound, int);
+      forall (i, o) in zip(retOffsets.domain, retOffsets) {
+        o = i * (n + 1);
+      }
+      var retBytes = makeDistArray(nFound * (n + 1), uint(8));
+      var srcInds = makeDistArray(nFound * (n + 1), int);
+      var dstInds = (+ scan longEnough) - longEnough;
+      ref oa = offsets.a;
+      if kind == Fixes.prefixes {
+        forall (d, o, le) in zip(dstInds, oa, longEnough) with (var agg = newDstAggregator(int)) {
+          if le {
+            const srcIndStart = d * (n + 1);
+            for j in 0..#(n+1) {
+              agg.copy(srcInds[srcIndStart + j], o + j);
+            }
+          }
+        }
+      } else if kind == Fixes.suffixes {
+        forall (d, o, l, le) in zip(dstInds, oa, lengths, longEnough) with (var agg = newDstAggregator(int)) {
+          if le {
+            const srcIndStart = d * (n + 1);
+            const byteStart = o + l - n;
+            for j in 0..#(n+1) {
+              agg.copy(srcInds[srcIndStart + j], byteStart + j);
+            }
+          }
+        }
+      }
+      ref va = values.a;
+      forall (b, i) in zip(retBytes, srcInds) with (var agg = newSrcAggregator(uint(8))) {
+        agg.copy(b, va[i]);
+      }
```

Comment:
I think this is the write way to do it with gathering the indices first

At first, I felt like we should be able to write directly into `retBytes`, but I convinced myself that this would cause remote-to-remote issues because it essentially boils down to
```chapel
agg.copy(retBytes[srcIndStart + j], va[byteStart + j]);
```

Reply:
Yeah, it feels like two `forall`s shouldn't be necessary, but I don't see a way to do it in one `forall` without remote-to-remote. This is similar to how we had to do it for the string gather implementation, `SegString.this(iv: [] int)`.